Fix #10059: limit auto initiated upload to 10 errors or cancellations

maxme opened the pull request:
Fix #10059: limit auto initiated upload to 10 errors or cancellations

Note that if the user retries 10 times manually, auto upload will be disabled as well. We're not counting if the error comes from an auto or manual action.

I marked this as a draft until the FluxC counter part is merged: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1296

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

malinajirka commented:
Thanks @maxme! It looks good overall - I've tested the behavior and reviewed the code 👍 . 

It seems the unit tests for the LocalDraftUploadStarter are missing the new constructor parameter. We might also want to consider adding unit tests for the new filter, wdyt?

maxme commented:
> It seems the unit tests for the LocalDraftUploadStarter are missing the new constructor parameter. We might also want to consider adding unit tests for the new filter, wdyt?

I fixed the tests c108dba  and added a new one in 4f22649
